### ISSUE TITLE
Cannot import importlib.resources with importlib-resources installed

### ISSUE THREAD

**Issue by Chiiizzzy** [python/importlib_resources#246]
I've tried to install `importlib-resources` v5.4/5.2/5.0/1.3/0.5 in python. But with all these packages, I cannot `import importlib.resources`, which throws `ModuleNotFoundError: No module named 'importlib.resources'`.

I'm using macOS menterey 12.2.1, intel, python3.6.9.

**Comment by hungrymonkey** [comment]
```
try:
    import importlib.resources as pkg_resources
except ImportError:
    import importlib_resources as pkg_resources
```
This library should not be imported when python is greater than 3.7. You can test if you need the external package or built-in library.

```setup.py
if sys.version_info < (3, 7):
    REQUIRES += [ "importlib_resources" ]
```
I hope this post helps.

**Comment by jaraco** [comment]
> This library should not be imported when python is greater than 3.7. You can test if you need the external package or built-in library.

Correction - this library supports Python 2.7+. The most recent releases only support later Python versions, but older releases are still viable and supported on older Pythons.

> You can test if you need the external package or built-in library.

A better way to declare a Python-specific dependency is to use environment markers.

```
install_requires =
  importlib_resources; python_version < "3.7"
```

**Comment by jaraco** [comment]
But it's correct. `importlib.resources` is not available until Python 3.7.
